File: 1157.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(){
    char *buffer= (char*)malloc(sizeof(char)*1000000+1);
    scanf("%s",buffer);
    int leng = strlen(buffer);
    char *input_str = (char*)malloc(sizeof(char)*leng+1);
    strcpy(input_str, buffer);
    free(buffer);
    int alpha[26] = {0,};

    for(int i=0;i<leng;i++){
        char tar= input_str[i];    
        if( /*tar<123&&*/tar>96){
            tar-=32;
        }

        if( tar>90||tar<65){
            printf("\nnot alphabet error\n ");
            return -1;
        }else{
            alpha[tar-65]++;
        }
    }
    
    char biggest = '?';
    int bignum = 0;

    for(int i=0;i<sizeof(alpha)/sizeof(int);i++){
        int num = alpha[i];
        char tar_char = i+65;
        
        if(bignum<num){
            biggest = tar_char;
            bignum=num;

        }else if(bignum==num){
            biggest = '?';
        }
    }
    printf("%c",biggest);
    free(input_str);
    return 0;
}

File: 1157.py
alpha = [0]*26
string = input()
string=string.upper()
#print(string)
for a in string:
    a=ord(a)-65
    #print(a)
    if(a<0 or a>25):
        print("not alphabet")
        quit()
    alpha[a]+=1
biggest = ["?",0]
for a in range(26):
    num = alpha[a]
    tar = chr(a+65)
    if   num > biggest[1]:
        biggest[0] = tar
        biggest[1]=num
    elif num == biggest[1]:
        biggest[0]="?"
print(biggest[0])
